pandas/core/frame.py

```diff
@@ -2891,7 +2891,10 @@ def _maybe_casted_values(index, labels=None):
                                                                 np.nan)
             return values
 
-        new_index = np.arange(len(new_obj), dtype='int64')
```

Comment:
``new_index = _default_index(len(new_obj))``
is the entire answer here, you don't need a separate check whether its a ``RangeIndex`` or not